azurelinuxagent/common/rdma.py

```diff
@@ -90,3 +96,133 @@ def reboot_system(self):
         shellutil.run('shutdown -r now')
+
+dapl_config_paths = [
+    '/etc/dat.conf', '/etc/rdma/dat.conf', '/usr/local/etc/dat.conf']
+
+class RDMADeviceHandler(object):
+
+    """
+    Responsible for writing RDMA IP and MAC address to the /dev/hvnd_rdma
+    interface.
+    """
+
+    rdma_dev = '/dev/hvnd_rdma'
+    device_check_timeout_sec = 120
+    device_check_interval_sec = 1
+
+    ipv4_addr = None
+    mac_adr = None
+
+    def __init__(self, ipv4_addr, mac_addr):
+        self.ipv4_addr = ipv4_addr
+        self.mac_addr = mac_addr
+
+    def start(self):
+        """
+        Start a thread in the background to process the RDMA tasks and returns.
+        """
+        logger.info("RDMA: starting device processing in the background.")
+        threading.Thread(target=self.process).start()
+
+    def process(self):
+        RDMADeviceHandler.update_dat_conf(dapl_config_paths, self.ipv4_addr)
+        RDMADeviceHandler.wait_rdma_device(
+            self.rdma_dev, self.device_check_timeout_sec, self.device_check_interval_sec)
+        RDMADeviceHandler.write_rdma_config_on_device(
+            self.rdma_dev, self.ipv4_addr, self.mac_addr)
+        RDMADeviceHandler.update_network_interface(self.mac_addr, self.ipv4_addr)
+
+    @staticmethod
+    def update_dat_conf(paths, ipv4_addr):
+        """
+        Looks at paths for dat.conf file and updates the ip address for the
+        infiniband interface.
+        """
+        logger.info("Updating DAPL configuration file")
+        for f in paths:
+            logger.info("RDMA: trying {0}".format(f))
+            if not os.path.isfile(f):
+                logger.info(
+                    "RDMA: DAPL config not found at {0}".format(f))
+                continue
+            logger.info("RDMA: DAPL config is at: {0}".format(f))
+            cfg = fileutil.read_file(f)
+            new_cfg = RDMADeviceHandler.replace_dat_conf_contents(
+                cfg, ipv4_addr)
+            fileutil.write_file(f, new_cfg)
+            logger.info("RDMA: DAPL configuration is updated")
+            return
+
+        raise Exception("RDMA: DAPL configuration file not found at predefined paths")
+
+    @staticmethod
+    def replace_dat_conf_contents(cfg, ipv4_addr):
+        old = "ofa-v2-ib0 u2.0 nonthreadsafe default libdaplofa.so.2 dapl.2.0 \"\S+ 0\""
+        new = "ofa-v2-ib0 u2.0 nonthreadsafe default libdaplofa.so.2 dapl.2.0 \"{0} 0\"".format(
+            ipv4_addr)
+        return re.sub(old, new, cfg)
+
+    @staticmethod
+    def write_rdma_config_on_device(path, ipv4_addr, mac_addr):
+        data = RDMADeviceHandler.get_rdma_config(ipv4_addr, mac_addr)
+        logger.info(
+            "RDMA: Updating device with configuration: {0}".format(data))
+        with open(path, "w") as c:
+            c.write(data)
+        logger.info("RDMA: Updated device with IPv4/MAC addr successfully")
+
+    @staticmethod
+    def get_rdma_config(ipv4_addr, mac_addr):
```

Comment:
I would say this is more of a "generate" method then a "get" method. The method does not get data from somewhere, it generates the data given into the proper format. I think "get_rdma_device_config" would be a more descriptive name. Also as the parameter name ip_addr is probably a more forward looking choice. It is only a matter of time before IPv6 will be supported and then the parameter has to be renamed and if it is forgotten it will be confusing.